### PR TITLE
[Fix](vec-stream-load-json) simdjson throw execption lead to core dump

### LINKED ISSUE #11879: [Bug] (vec-stream-load-json) simdjson throw execption lead to core dump
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?

when config::enable_simdjson_parser=true in vec streamload, may lead to core dump when json input invalid format string like `{ "a`,  or all the fields is null like '{}', this may lead to simdjson lib throw some unhandled expection like `Objects and arrays can only be iterated when they are first encountered`

### What You Expected?

no core

### How to Reproduce?

input some invalid json format string 

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by eldenmoon**
when config::enable_simdjson_parser=true in vec streamload, may lead to core dump when json input invalid format string like `{ "a'` , or all the fields is null like `'{}'`, this may lead to simdjson lib throw some unhandled expection like `Objects and arrays can only be iterated when they are first encountered`.We should take care of these cases

# Proposed changes

Issue Number: close #11879

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/exec/vjson_scanner.cpp [hunk 1]
```diff
@@ -627,16 +633,17 @@ Status VSIMDJsonReader::_set_column_value(simdjson::ondemand::value objectValue,
...
         return Status::OK();
     }
 
     auto object_val = objectValue.get_object();
     size_t cur_row_count = block.rows();
     // iterate through object, simdjson::ondemond will parsing on the fly
     for (auto field : object_val) {
-        std::string key(field.unescaped_key().value());
-        auto column_type_and_name = block.try_get_by_name(key);
+        std::string_view key;
```
Comment (stalary) [inline be/src/vec/exec/vjson_scanner.cpp:644]: Key doesn't seem to be initialized?

Comment (eldenmoon) [reply be/src/vec/exec/vjson_scanner.cpp:644]: field.unescaped_key().get(key) will initialize key，which is get(T& )

Comment (stalary) [reply be/src/vec/exec/vjson_scanner.cpp:644]: ok, thanks~

**Review by stalary — APPROVED** [review]
LGTM

**Comment by Gabriel39** [comment]
Passed all P0 regression tests after this PR?

**Comment by yiguolei** [comment]
> Passed all P0 regression tests after this PR?

Seems yes....

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**